nav2_smac_planner/src/node_lattice.cpp

```diff
@@ -472,14 +480,7 @@ void NodeLattice::getNeighbors(
   Coordinates initial_node_coords, motion_projection;
   MotionPrimitivePtrs motion_primitives = motion_table.getMotionPrimitives(this);
   const float & grid_resolution = motion_table.lattice_metadata.grid_resolution;
-
-  unsigned int direction_change_idx = 1e9;
-  for (unsigned int i = 0; i != motion_primitives.size(); i++) {
-    if (motion_primitives[0]->start_angle != motion_primitives[i]->start_angle) {
-      direction_change_idx = i;
-      break;
-    }
-  }
+  unsigned int start_angle_index = static_cast<unsigned int>(motion_primitives[0]->start_angle);
```

Comment:
```suggestion
  unsigned int direction_change_idx = motion_table.direction_change_indices[motion_primitives[0]->start_angle];
```

Then use `direction_change_idx` in L506. That prevents an additional lookup every loop. This is a minor optimization, but relative to the speed ups removing this loop will provide, its important

Reply:
Can you give me a link to that patch? Either would be technically correct. I think I'd prefer the `getMotionprimitives` version, but I'd want to see if before saying that definitely

Reply:
Just pushed it to temporary branch here: https://github.com/akchobby/navigation2/commits/tmp-direction-change/

Reply:
Yeah lets go with this instead